core/server/data/migrations/versions/3.23/01-migrate-bulk-email-settings.js

```diff
@@ -0,0 +1,64 @@
+const logging = require('../../../../../shared/logging');
+
+module.exports = {
+    config: {
+        transaction: true
+    },
+
+    async up({transacting: knex}) {
+        const defaultValue = {
+            provider: 'mailgun',
+            apiKey: '',
+            domain: '',
+            baseUrl: ''
+        };
+        const bulkEmailSettingsJSON = await knex('settings').where({
+            key: 'bulk_email_settings'
+        }).select().first();
+
+        let bulkEmailSettings;
+        try {
+            bulkEmailSettings = JSON.parse(bulkEmailSettingsJSON.value) || defaultValue;
+        } catch (err) {
+            logging.warn(`Error parsing bulk_email_settings JSON. Using defaults`);
+            bulkEmailSettings = defaultValue;
+        }
+
+        const operations = [{
+            key: 'mailgun_api_key',
+            value: bulkEmailSettings.apiKey
+        }, {
+            key: 'mailgun_domain',
+            value: bulkEmailSettings.domain
+        }, {
+            key: 'mailgun_base_url',
+            value: bulkEmailSettings.baseUrl
+        }];
+
+        for (const operation of operations) {
+            logging.info(`Updating ${operation.key} setting's value, group, type & flags.`);
+            await knex('settings')
+                .where({
+                    key: operation.key
+                })
+                .update({
+                    group: 'email',
+                    type: 'string',
+                    flags: '',
```

Comment:
Guess doesn't make much of a difference, but probably better to set `flags` here as `null` (or maybe just leave it out in update) instead of empty string as looking at DB thats consistent with how rest of the empty `flags` value is setup, which are all null :)